File: src/main/java/com/zerozero/store/infrastructure/kakao/search/request/KakaoSearchRequest.java
package com.zerozero.store.infrastructure.kakao.search.request;

import com.zerozero.store.infrastructure.kakao.search.core.CategoryGroupCode;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class KakaoSearchRequest {
    private String query;
    private CategoryGroupCode categoryGroupCode;
    private String x;
    private String y;
    private Integer radius;
    private String rect;
    private Integer page;
    private Integer size;
    private String sort;
}

File: src/main/java/com/zerozero/store/infrastructure/kakao/search/request/KakaoSearchRequest.kt
package com.zerozero.store.infrastructure.kakao.search.request

import com.zerozero.store.infrastructure.kakao.search.core.CategoryGroupCode

data class KakaoSearchRequest(
    val query: String,
    val categoryGroupCode: CategoryGroupCode? = null,
    val x: String? = null,
    val y: String? = null,
    val radius: Int? = null,
    val rect: String? = null,
    val page: Int = 1,
    val size: Int = 15,
    val sort: String = "accuracy",
)
